### PR TITLE
Hash - add additional hash key lengths

### PR DESCRIPTION
**Pull request by jeremylt**
This is a small PR to add some additional hash key sizes to help me improve integration on the MFEM side.

### CODE REVIEW

File: include/ceed-hash.h [hunk 1]
```diff
@@ -67,6 +67,43 @@ static inline CeedHash_t CeedHashCombine(CeedHash_t seed, CeedHash_t hash) {
   return seed ^ (hash + (seed << 6) + (seed >> 2));
 }
 
+// IJ - two keys
+typedef struct _CeedHashIJKey { CeedInt i, j; } CeedHashIJKey;
+#define CeedHashIJKeyHash(key) \
+  CeedHashCombine(CeedHashInt((key).i),CeedHashInt((key).j))
+
+#define CeedHashIJKeyEqual(k1,k2) \
+  (((k1).i==(k2).i) ? ((k1).j==(k2).j) : 0)
```
Comment (jedbrown) [inline include/ceed-hash.h:76]:
```suggestion
  ((k1).i==(k2).i && (k1).j==(k2).j)
```

File: include/ceed-hash.h [hunk 2]
```diff
@@ -67,6 +67,43 @@ static inline CeedHash_t CeedHashCombine(CeedHash_t seed, CeedHash_t hash) {
...
+	KHASH_INIT(name,CeedHashIJKey,value,1,CeedHashIJKeyHash,CeedHashIJKeyEqual)
+
+typedef struct _CeedHashIJKKey { CeedInt i, j, k; } CeedHashIJKKey;
+#define CeedHashIJKKeyHash(key) \
+  CeedHashCombine(CeedHashCombine(CeedHashInt((key).i),CeedHashInt((key).j)), \
+                  CeedHashInt((key).k))
+
+// IJK - three keys
+#define CeedHashIJKKeyEqual(k1,k2) \
+  (((k1).i==(k2).i) ? ((k1).j==(k2).j) ? ((k1).k==(k2).k) : 0 : 0)
```
Comment (jedbrown) [inline include/ceed-hash.h:88]: And here.

File: include/ceed-hash.h [hunk 3]
```diff
@@ -67,6 +67,43 @@ static inline CeedHash_t CeedHashCombine(CeedHash_t seed, CeedHash_t hash) {
...
+#define CeedHashIJKInit(name, value)										\
+	KHASH_INIT(name,CeedHashIJKKey,value,1,CeedHashIJKKeyHash,CeedHashIJKKeyEqual)
+
+// IJKL - four keys
+typedef struct _CeedHashIJKLKey { CeedInt i, j, k, l; } CeedHashIJKLKey;
+#define CeedHashIJKLKeyHash(key) \
+  CeedHashCombine(CeedHashCombine(CeedHashInt((key).i),CeedHashInt((key).j)), \
+                  CeedHashCombine(CeedHashInt((key).k),CeedHashInt((key).l)))
+
+#define CeedHashIJKLKeyEqual(k1,k2) \
```
Comment (jedbrown) [inline include/ceed-hash.h:99]: and here

File: include/ceed-hash.h (hunk 1 above)
Comment (jeremylt) [reply include/ceed-hash.h]: Updated (x4)

**Pull request merged**